river/command.zig

```diff
@@ -117,3 +117,59 @@ pub fn errToMsg(err: Error) [:0]const u8 {
         Error.Other => unreachable,
     };
 }
+
+/// Parses a list of options specified by T
+/// args[0] must be the first option if any
+/// offset will be set to the index of the first entry in args that is not an option
+/// See test "optinal args" for an example of T
+pub fn parse(args: []const []const u8, offset: *usize, comptime T: type) T {
+    // The empty {} make sure there is a default value for all members
+    var ret: T = T{};
+    // Make sure the format of T is valid
+    if (@typeInfo(T) != .Struct) @compileError("T must be a struct");
+    inline for (@typeInfo(T).Struct.fields) |field| {
+        if (@typeInfo(field.field_type) != .Bool)
+            @compileError("unsupported option type " ++ field.field_type);
+        if (field.default_value.?)
+            @compileError("bool options must be false by default");
+        if (field.name[0] != '-')
+            // TODO Should we really have this check? If yes test ascii.isAlpha(field.name[1])
+            // Figure out how long arguments should be is -a a valid argument
+            @compileError("option names must start with a single - followed by an alphabetical cahr");
```

Comment:
I'd say just drop this check. Options not conforming to the naming convention will be fairly easy to catch in review.